hack/kind-test.sh

```diff
@@ -25,57 +25,37 @@ REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 TAG_NAME=$(date "+gmp-%Y%d%m_%H%M")
 TEST_ARGS=""
 # Convert kind cluster name to required regex if necessary.
-KIND_CLUSTER=`echo ${GO_TEST} | sed -r 's/([A-Z])/-\L\1/g' | sed 's/^-//'`
+KIND_CLUSTER=$(echo ${GO_TEST} | sed -r 's/([A-Z])/-\L\1/g' | sed 's/^-//')
 KUBECTL="kubectl --context kind-${KIND_CLUSTER}"
 # Ensure a unique label on any test data sent to GCM.
 GMP_CLUSTER=$TAG_NAME
-if [ -f "$GOOGLE_APPLICATION_CREDENTIALS" ]; then
-  PROJECT_ID=$(jq -r '.project_id' ${GOOGLE_APPLICATION_CREDENTIALS})
+if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+  PROJECT_ID=$(jq -r '.project_id' "${GOOGLE_APPLICATION_CREDENTIALS}")
 else
   echo ">>> no credentials specified. running without GCM validation"
   TEST_ARGS="${TEST_ARGS} -skip-gcm"
-  # Create mock credentials.
```

Comment:
Interesting - this bug may have been solved at some point then.

Before, even if you passed a disable-auth flag to the collector, it would still crashloop with this error message without some form of credentials.